Algorithms and Data Structures/JasonConvertExercise/JasonConvertExercise/Program.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace JasonParserExercise
+{
+    internal class Program
+    {
+        public static void Main(string[] args)
+        {
+            Parse();
+        }
+
+        private static void Parse()
+        {
+            try
+            {
+                string playerDataJson = Console.ReadLine();
+                var playerDataDeserializer = new PlayerDataDeserializer();
+                var convertedPlayerData = playerDataDeserializer.DeserializePlayerData(playerDataJson);
+
+                if (convertedPlayerData.username == null)
+                {
+                    Console.WriteLine("username is a necessary field and cannot be null!");
+                    Parse();
+                }
+                else if (convertedPlayerData.level == 0)
+                {
+                    convertedPlayerData.level = 1;
+                }
```

Comment:
the default values should be handled better, think about we want to have multiple default values for each field, multiple ifs is code smell and should indicate that you have to find a better way to solve this problem